langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Matches.java

```diff
@@ -36,8 +37,9 @@ public static BArray matchAt(BRegexpValue regExp, BString str, int startIndex) {
         checkIndexWithinRange(str, startIndex);
         Matcher matcher = RegexUtil.getMatcher(regExp, str);
         matcher.region(startIndex, str.length());
```

Comment:
Don't we need to add 
```
int adjustedStartIndex = getSurrogateAdjustedStartIndex((int) startIndex, surrogates);
```
in here.

Reply:
This is currently failing

```
import ballerina/lang.regexp;
import ballerina/io;

public function main() {
    string str18 = "🔜 #R";
    regexp:Span? result18_1 = re `#R`.matchAt(str18, 2);
    io:println((<regexp:Span> result18_1).substring(), " ", (<regexp:Span> result18_1).startIndex, " ", (<regexp:Span> result18_1).endIndex);
}   
```

Reply:
Seems there are couple of other places as well eg: findAll. Will update

Reply:
Now it is working
tested for find Functions, match functions